autogpt_platform/frontend/src/app/auth/integrations/oauth_callback/route.ts

```diff
@@ -19,13 +20,15 @@ export async function GET(request: Request) {
           message: `Incomplete query: ${searchParams.toString()}`,
         };
 
+  console.debug("Sending message to opener:", message);
+
   // Return a response with the message as JSON and a script to close the window
   return new NextResponse(
     `
     <html>
       <body>
         <script>
-          window.postMessage(${JSON.stringify(message)});
+          window.opener.postMessage(${JSON.stringify(message)});
```

Comment:
?

Reply:
I purposefully used `window` for this and not `window.opener` because it has a lot less going on, and because the need for the communication channel is tied to the existence of the window itself.

Reply:
so I remember why I added this

It doesn't ever tell the main script that its done without it on edge mac

Reply:
So MS Edge on macOS doesn't properly implement window messaging buses? F

Reply:
its a life to live for sure